protocols/kad/src/kbucket/key.rs

```diff
@@ -169,6 +169,13 @@ impl AsRef<KeyBytes> for KeyBytes {
 #[derive(Copy, Clone, PartialEq, Eq, Default, PartialOrd, Ord, Debug)]
 pub struct Distance(pub(super) U256);
 
+impl Distance {
+    /// Returns the base 2 logarithm of the [`Distance`].
+    pub fn log2(&self) -> Option<u32> {
+        (256 - self.0.leading_zeros()).checked_sub(1)
```

Comment:
Instead of the hard-coded `256` here, we could as well derive it. But we would then depend on the internal representation of a `U256` instead. Let me know what you think.

```suggestion
        (U256::MAX.bits() - self.0.leading_zeros()).checked_sub(1)
```

Reply:
Either way sounds fine to me.

Reply:
Leaving as is then.